### PR TITLE
Upgrade to poem 3.x.x and poem-openapi 5.x.x

### PR DESCRIPTION
**Pull request by banool**
## Description
In this PR I upgrade us from poem 1.x.x and poem-openapi 2.x.x to poem 3.x.x and poem-openapi 5.x.x. Besides it being good in general to stay up to date, we want to upgrade specifically to get the latest opentelemetry features so we can benefit from distributed tracing throughout the API stack. There have been other instances where being pinned to a specific version caused us issues.

The new Poem traits had issues with `Box<UserTransaction>` in transaction, so I just unboxed it. I can't see why it needs to be in a Box and the original commit (https://github.com/aptos-labs/aptos-core/commit/3ecd8e927a35ead89f528f035c0c6397ffb8d80b) doesn't explain it. Perhaps there used to be some recursive type that we had to break.

Besides that, fortunately there wasn't much to change with our types or our custom derive macros.

By default this changes the spec slightly, primarily the inclusion of `charset=utf-8` in the content type. I recall this caused issues for code generators, and it'd require changes to our Accept logic, so I remove that again. It also seems to change some of the enum variants. I don't believe we make use of the OpenAPI spec anywhere anymore so it shouldn't affect us, but we should reach out to anyone who might be affected (perhaps an old Go SDK, if it's still in development).

I will wait for https://github.com/poem-web/poem/pull/829 to be released and then upgrade the poem deps again before releasing this PR.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [x] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [x] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
For now just CI, let's see how that goes first.

I run a localnet and connected the explorer to it and clicked around and everything seemed to work fine, no errors from the network tab. The spec explorer (at /v1/spec) works as before.

## Key Areas to Review
- Ensure `Box<UserTransaction>` -> `UserTransaction` is valid.
- Make sure the modified OpenAPI spec is okay.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by geekflyer — APPROVED** [review]

**Comment by banool** [comment]
Applying the do not merge label until https://github.com/poem-web/poem/pull/829 is released.

**Review by JoshLind — APPROVED** [review]

**Comment by banool** [comment]
Okay the maintainer seems to be away, we'll just use a git dep for now.

**Comment by banool** [comment]
I confirmed that a localnet and the spec explorer work still, let's send it.

**Pull request merged**
